kibana-reports/public/components/report_definitions/report_settings/report_settings.tsx

```diff
@@ -658,6 +680,19 @@ export function ReportSettings(props: ReportSettingProps) {
         setDefaultEditValues(editData, reportSourceOptions);
       }
     });
+
+    if (window.location.href.indexOf("?") > -1) {
+      const url = window.location.href;
+      const reportSourceInfo = url.split("?");
+      const id = reportSourceInfo[1].substring(
+        reportSourceInfo[1].indexOf(":") + 1, 
+        reportSourceInfo[1].length
+      );
```

Comment:
suggest to refactor to a reusable method

urlParse(url) {
  return {} // a parsed object
}

Reply:
Will do